app/models/archived_patient.rb

```diff
@@ -0,0 +1,146 @@
+  field :appointment_date, type: Date
+  field :procedure_cost, type: Integer
+  field :patient_contribution, type: Integer
+  field :naf_pledge, type: Integer
+  field :fund_pledge, type: Integer
+  field :pledge_sent, type: Boolean
+  field :resolved_without_fund, type: Boolean
+  field :pledge_generated_at, type: Time
+  field :pledge_sent_at, type: Time
+
+  # Indices
+  index(line: 1)
+  index(identifier: 1)
+
+  # Validations
+  validates :initial_call_date,
+            :created_by_id,
+            :line,
+            presence: true
+  validates :appointment_date, format: /\A\d{4}-\d{1,2}-\d{1,2}\z/,
+                               allow_blank: true
+
+  validates_associated :fulfillment
+
+
+  mongoid_userstamp user_model: 'User'
+
+  # Archive & Delete patients who have fallen out of contact
+  def self.archive_dropped_off_patients!
+    Patient.where(:initial_call_date.lte => 1.year.ago).each do |patient|
+      ArchivedPatient.convert_patient(patient)
+      patient.destroy!
+    end
+  end
+
+  # Archive & Delete patients who are completely through our process
+  def self.archive_fulfilled_patients!
+    Patient.fulfilled_on_or_before(120.days.ago) do |patient| ## TODO get accountant signoff on 120 days?
+      ArchivedPatient.convert_patient(patient)
+      patient.destroy!
+    end
+  end
+
+  def self.convert_patient(patient)
+    archived_patient = create(
+      line: patient.line,
+      city: patient.city,
+      state: patient.state,
+      county: patient.county,
+      initial_call_date: patient.initial_call_date,
+      appointment_date: patient.appointment_date,
+
+      urgent_flag: patient.urgent_flag,
+      referred_by: patient.referred_by,
+      referred_to_clinic: patient.referred_to_clinic,
+
+      last_menstrual_period_weeks: patient.last_menstrual_period_weeks,
+      last_menstrual_period_days: patient.last_menstrual_period_days,
+
+      race_ethnicity: patient.race_ethnicity,
+      employment_status: patient.employment_status,
+      insurance: patient.insurance,
+      income: patient.income,
+      language: patient.language,
+      voicemail_preference: patient.voicemail_preference,
+
+      procedure_cost: patient.procedure_cost,
+      patient_contribution: patient.patient_contribution,
+      naf_pledge: patient.naf_pledge,
+      fund_pledge: patient.fund_pledge,
+
+      pledge_sent: patient.pledge_sent,
+      resolved_without_fund: patient.resolved_without_fund,
+      pledge_generated_at: patient.pledge_generated_at,
+      pledge_sent_at: patient.pledge_sent_at,
+
+      pledge_generated_by: patient.pledge_generated_by,
+      pledge_sent_by: patient.pledge_sent_by,
+      created_by_id: patient.created_by_id,
+
+      age_range: patient.age_range,
+      has_alt_contact: patient.determine_has_other_contact,
+      notes_count: patient.notes_count,
+      has_special_circumstances: patient.has_special_circumstances,
+
+    )
+    archived_patient.save! # Do I need this?
```

Comment:
I think I see your instinct here. Given that we want to roll this back if any part of this fails, maybe instead of `create`s we go with `new` and then do an `archived_patient.save!` at the end? This is kind of cheating our way into a database transaction but I think it will work. Thoughts?